app/src/org/commcare/activities/components/ImageCaptureProcessing.java

```diff
@@ -117,10 +136,49 @@ public static void processImageChooserResponse(FormEntryActivity activity,
 
         // get gp of chosen file
         Uri selectedImage = intent.getData();
-        String imagePath = FileUtil.getPath(activity, selectedImage);
 
+        if (selectedImage == null) {
+            showInvalidImageMessage(activity);
+            return;
+        }
+
+        try {
+            String imagePath = UriToFilePath.getPathFromUri(activity, selectedImage);
+            processImageGivenFilePath(activity, instanceFolder, imagePath);
+        } catch (UriToFilePath.NoDataColumnForUriException e) {
+            // Can't get file path from Uri, so need to work with uri instead
+            processImageGivenFileUri(activity, instanceFolder, selectedImage);
```

Comment:
Handling uri differently than filepath (If we can't get gile path, we are using the uri to first make a copy of the file internally and then passing that new file's path to the `processImageGivenFilePath`)

Reply:
When choosing a file from the file system this code looks like it ends up trying to delete the selected image from its original location, which doesn't seem like the right approach to me. If someone just _Took_ a picture it's fine (desirable actually) for it to not exist elsewhere on the device. If they chose one, they'll likely be annoyed if it gets deleted afterwards.